cluster/write-file.go

```diff
@@ -0,0 +1,68 @@
+package cluster
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"github.com/docker/docker/api/types/container"
+	"github.com/rancher/rke/docker"
+	"github.com/rancher/rke/hosts"
+	"github.com/rancher/rke/log"
+	"github.com/sirupsen/logrus"
+	"path/filepath"
+)
+
+// WriteFile - Writes file to Node.
+func (c *Cluster) WriteFile(ctx context.Context, host *hosts.Host, role string, path string, mode string, content string) error {
+	log.Infof(ctx, "[%s] Deploying file %s on node [%s]", role, path, host.Address)
+	err := c.doWriteFile(ctx, host, role, path, mode, content)
+	if err != nil {
+		return fmt.Errorf("[%s] Failed to file %s on node [%s]: %v", role, path, host.Address, err)
+	}
+	logrus.Debugf("[%s] Successfully deployed %s on node [%s]", role, path, host.Address)
+
+	return nil
+}
+
+func (c *Cluster) doWriteFile(ctx context.Context, host *hosts.Host, role string, path string, mode string, content string) error {
```

Comment:
there is an already function that does this https://github.com/rancher/rke/blob/master/cluster/file-deployer.go#L32, i think this should be reused

Reply:
Do you mind if I enhance file-deployer? 
I think its cleaner to use base64 encoding (so no escaping issues) and create a backup copy before writing over an existing file. Also most of these files should probably not be 0644 (or at least not world readable) so mode should be configurable.

Reply:
that's totally fine but can you combine both of these methods